### ISSUE TITLE
[MCS-DE-GO] Trigger "Condition" property syntax validation not working in all cases

### ISSUE THREAD

**Issue by goergenj** [microsoft/BotFramework-Composer#5610]
When defining a "Condition" on an Intent Trigger, the syntax validation only works when the condition is prefixed with an "=".

Example Condition "#Intentname.score > 0.6".

If you do not add an "=" at the beginning the condition still works, however the syntax validation is not active in this case.

![image](https://user-images.githubusercontent.com/31212469/105860666-1b4fc580-5fee-11eb-85fc-5ded169ace3a.png)

This would cause however a corrupt LUIS model as the LU-File would have an error.

If you add "=" syntax validation works as expected

![image](https://user-images.githubusercontent.com/31212469/105860761-3a4e5780-5fee-11eb-8d49-1bba67bec80f.png)

![image](https://user-images.githubusercontent.com/31212469/105860792-42a69280-5fee-11eb-936a-ab885cc878d2.png)

Either the syntax validation should always be active independent of the "=" prefix and/or a missing "=" should trigger a syntax validation error.

@joaollq @goergenj @garypretty @benbrown @cwhitten

**Comment by tdurnford** [comment]
@goergenj Are you still having an issue with this? I'm not able to reproduce it in the latest build from main. In the sdk schema, it looks like a condition is an `expression` and not an `equalsExpression`, so you don't need to preface it with an equals sign.

```
"condition": {
  "$role": "expression",
  "title": "Boolean condition",
  "description": "Boolean constant or expression to evaluate.",
  "oneOf": [
    {
      "$ref": "#/definitions/expression"
    },
    {
      "type": "boolean",
      "title": "Boolean",
      "description": "Boolean value.",
      "default": true,
      "examples": [
        false
      ]
    }
  ]
},
```

**Comment by goergenj** [comment]
@tdurnford Not sure we are on the same page. I was referring to the syntax checker only. The Condition itself works. I can recheck with 1.4. and let you know, if it's fixed now.

**Comment by tdurnford** [comment]
@tomlm Can you weigh in on why users need to use the equals sign in some cases and not in others? 

I believe it’s because if the property type is not a string - number, boolean, etc. - and the property value is a string, the BotFramework will assume the value is an expression that resolves to the expected property type. In this case, the equals sign for an expression is not needed. 

Users only need to use the equals sign if the expected value can be a string so that the framework knows it’s an expression that resolves to some type and not just a string. 

I do agree with @goergenj - it does lead to confusion especially in the UI Context.

**Comment by tdurnford** [comment]
@goergenj can you check if this is resolved if 2.0 that was just released last week?

**Comment by goergenj** [comment]
@tdurnford With v2.0 the syntac validation seems to be completely turned off now. I can add extraneuous carachters now with or withour "=" and never get an error. Is this the expected behaviour now not having syntac validation at this field?
